app/src/main/java/com/hedvig/app/feature/embark/ui/EmbarkActivity.kt

```diff
@@ -27,26 +26,20 @@ import com.hedvig.app.feature.embark.passages.textactionset.TextActionSetFragmen
 import com.hedvig.app.feature.embark.passages.textactionset.TextActionSetParameter
 import com.hedvig.app.util.extensions.view.remove
 import com.hedvig.app.util.extensions.viewBinding
-import e
 import org.koin.android.viewmodel.ext.android.viewModel
 
 class EmbarkActivity : BaseActivity(R.layout.activity_embark) {
     private val model: EmbarkViewModel by viewModel()
     private val binding by viewBinding(ActivityEmbarkBinding::bind)
 
+    private val storyName: String by lazy {
+        // TODO: Implement error UI that design must provide
+        intent.getStringExtra(STORY_NAME) ?: throw IllegalArgumentException("Programmer error: STORY_NAME not provided to ${this.javaClass.name}")
+    }
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
-        val storyName = intent.getStringExtra(STORY_NAME)
-
-        if (storyName == null) {
```

Comment:
What is the reason we are checking the parameter in oncreate? I think it makes more sense to check this before starting the embark activity and showing an error dialog in the previous activity.

Reply:
Just generally defensive programming, I think 🤔